clang/lib/Driver/Driver.cpp

```diff
@@ -696,6 +696,11 @@ static llvm::Triple computeTargetTriple(const Driver &D,
       Target.setEnvironment(llvm::Triple::GenericJS);
   }
 
+  // Force a full triple when targeting CheerpOS
+  if (Target.getVendor() == llvm::Triple::CheerpOS) {
+    Target.setEnvironment(llvm::Triple::WebAssembly);
```

Comment:
so internally the wasm32-cheerpos-linux becomes cheerp-cheerpos-wasm?

Reply:
Yes, correct, this solves the problem of routing to our custom backend when using LLVM-level tools (llc/opt). Does this feel acceptable?

Reply:
sounds good